resource/file/file.go

```diff
@@ -0,0 +1,566 @@
+// Copyright © 2016 Asteris, LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package file
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/user"
+	"path/filepath"
+	"strconv"
+
+	"github.com/asteris-llc/converge/resource"
+	"github.com/pkg/errors"
+)
+
+const defaultPermissions = os.FileMode(0750)
+
+// Action is the calculated action
+type Action uint8
+
+const (
+	// ActionNone indicates no action is required
+	ActionNone Action = iota
+
+	// ActionCreate creates a File resource
+	ActionCreate
+
+	// ActionModify deletes a File resource
+	ActionModify
+
+	// ActionDelete deletes a File resource
+	ActionDelete
+)
+
+func (a Action) String() string {
+	return string(a)
```

Comment:
this will return a number, which won't mean anything to the user. I'd suggest:

```go
switch a {
  case 0:
    return "none"

  case 1:
    return "create"

  // et cetera
}
```